pylate/evaluation/nano_beir_evaluator.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import annotations
+
+import logging
+from typing import Literal
+
+from sentence_transformers.evaluation.SentenceEvaluator import SentenceEvaluator
+from sentence_transformers.util import is_datasets_available
+
+from .pylate_information_retrieval_evaluator import PyLateInformationRetrievalEvaluator
+
+logger = logging.getLogger(__name__)
+
+DatasetNameType = Literal[
+    "climatefever",
+    "dbpedia",
+    "fever",
+    "fiqa2018",
+    "hotpotqa",
+    "msmarco",
+    "nfcorpus",
+    "nq",
+    "quoraretrieval",
+    "scidocs",
+    "arguana",
+    "scifact",
+    "touche2020",
+]
+
+
+dataset_name_to_id = {
+    "climatefever": "zeta-alpha-ai/NanoClimateFEVER",
+    "dbpedia": "zeta-alpha-ai/NanoDBPedia",
+    "fever": "zeta-alpha-ai/NanoFEVER",
+    "fiqa2018": "zeta-alpha-ai/NanoFiQA2018",
+    "hotpotqa": "zeta-alpha-ai/NanoHotpotQA",
+    "msmarco": "zeta-alpha-ai/NanoMSMARCO",
+    "nfcorpus": "zeta-alpha-ai/NanoNFCorpus",
+    "nq": "zeta-alpha-ai/NanoNQ",
+    "quoraretrieval": "zeta-alpha-ai/NanoQuoraRetrieval",
+    "scidocs": "zeta-alpha-ai/NanoSCIDOCS",
+    "arguana": "zeta-alpha-ai/NanoArguAna",
+    "scifact": "zeta-alpha-ai/NanoSciFact",
+    "touche2020": "zeta-alpha-ai/NanoTouche2020",
+}
+
+dataset_name_to_human_readable = {
+    "climatefever": "ClimateFEVER",
+    "dbpedia": "DBPedia",
+    "fever": "FEVER",
+    "fiqa2018": "FiQA2018",
+    "hotpotqa": "HotpotQA",
+    "msmarco": "MSMARCO",
+    "nfcorpus": "NFCorpus",
+    "nq": "NQ",
+    "quoraretrieval": "QuoraRetrieval",
+    "scidocs": "SCIDOCS",
+    "arguana": "ArguAna",
+    "scifact": "SciFact",
+    "touche2020": "Touche2020",
+}
+
+
+class NanoBEIREvaluator(SentenceEvaluator):
+    """
+    This class evaluates the performance of a SentenceTransformer Model on the NanoBEIR collection of datasets.
+
+    The collection is a set of datasets based on the BEIR collection, but with a significantly smaller size, so it can be used for quickly evaluating the retrieval performance of a model before commiting to a full evaluation.
+    The datasets are available on HuggingFace at https://huggingface.co/collections/zeta-alpha-ai/nanobeir-66e1a0af21dfd93e620cd9f6
+    The Evaluator will return the same metrics as the InformationRetrievalEvaluator (i.e., MRR, nDCG, Recall@k), for each dataset and on average.
+
+
+    Example:
```

Comment:
This exemple docstring format won't be parsed by the documentation, you can use the standard format. Maybe we could use the example as unit test and rather import pylate